tests/integration_tests/test_mlflow.py

```diff
@@ -210,3 +212,162 @@ def test_tag_study_user_attrs(tmpdir: py.path.local, tag_study_user_attrs: bool)
         assert all((r.data.tags["my_study_attr"] == "a") for r in runs)
     else:
         assert all(("my_study_attr" not in r.data.tags) for r in runs)
+
+
+def test_study_name_with_decorator(tmpdir: py.path.local) -> None:
+
+    tracking_file_name = "file:{}".format(tmpdir)
+    study_name = "my_study"
+    n_trials = 3
+
+    metric_name = "additional_metric"
+    metric = 3.14
+
+    mlfli = MLFlowIntegrator(tracking_uri=tracking_file_name)
+
+    @track_in_mlflow(mlfli)
+    def _objective_func(trial: optuna.trial.Trial) -> float:
+
+        x = trial.suggest_float("x", -1.0, 1.0)
+        y = trial.suggest_float("y", 20, 30, log=True)
+        z = trial.suggest_categorical("z", (-1.0, 1.0))
+        assert isinstance(z, float)
+        trial.set_user_attr("my_user_attr", "my_user_attr_value")
+        mlflow.log_metric(metric_name, metric)
```

Comment:
Thank you for including this test, which is helpful to understand the usage of the integrator.


Ideally, I would like to use only `mlfli`: an instance of `MLFlowIntegrator`. What do you think about the following changes?

If we update the definition of `MLFlowIntegrator.log_metric` as fllows,

```python
    def log_metric(self, value: float, metric_name=None) -> None:
        trial_value = value if value is not None else float("nan")
        if metric_name is None:
            metric_name = self._metric_name
        mlflow.log_metric(metric_name, trial_value)
```
the reported value is still stored as `_metric_name` by default.  When we store an additional metric, we can store it by calling `mlfli.log_metric(metric_name, metric)`.

The drawback of this update is that the order of argument is swapped as in the `mlflow.log_metric`, I'm completely happy with this. so I would like to hear the opinion from you and other developers.

Reply:
I would like to avoid to integrate all possible functionalities of logging of MLFlow into the MLFlowIntegrator. Just out of my mind there are
- log_metric(metric_name, metric_value, step=None) and log_metrics
- log_param and log_params
- log_figure
- log_text
- log_artifact
- log_model
I would hence rather provide a simple functionality to use the features of MLFlow in Optuna without wrapping them all into an other class.

I would actually rather go and hide the MLFlowIntegrator from the User than extend it more :wink: 

But I am happy to consider an other opinion.

Reply:
I agree with you! Optuna does not need to provide all `log` functions in MLflow.